### PR TITLE
Solved level4

### PR DESCRIPTION
**Pull request by iuliagatea**

### CODE REVIEW

File: ruby_proj/app/bookstore.rb [hunk 1]
```diff
@@ -17,13 +17,14 @@ def sort!
 
   def add_to_cart(title)
     check_item = cart.find_item(title)
+    book = find_by(title)
```
Comment (paucio) [inline ruby_proj/app/bookstore.rb]: When searching for a book shouldn't we check if the book is in stock before adding it in the cart?

Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: I think it won't work for the inferior levels because there is no stock there.

Comment (asarbu) [reply ruby_proj/app/bookstore.rb]: maybe you should wrap this inside an `if` and check the stock of the book before adding it to the cart

Comment (paucio) [reply ruby_proj/app/bookstore.rb]: I think it's not right to add a book to cart if the stock is 0.

Maybe validate if found book respond to `stock` in the `find_by` method. If it does then check if the book is in stock if it doesn't respond then simply return the found book and it should work for the lower levels.

Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: How about this? 
```
def add_to_cart(title)
    check_item = cart.find_item(title)
    book = find_by(title)
    if book.stock.nil? || book.stock > 0
      if check_item.nil?
        item = Item.new(cart.item_count, book)
        cart.items << item
      else
        check_item.qty += 1
      end
      book.stock -= 1 if book.stock
    end
  end
```

Comment (paucio) [reply ruby_proj/app/bookstore.rb]: This method is getting kinda heavy... plz extract the inside if into a private method.

Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: what name do you suggest for the new method? `add_item` it's ok?

Comment (paucio) [reply ruby_proj/app/bookstore.rb]: maybe `add_item_to_cart`

File: ruby_proj/app/bookstore.rb [hunk 2]
```diff
@@ -17,12 +17,9 @@ def sort!
...
   def add_to_cart(title)
     check_item = cart.find_item(title)
-    if check_item.nil?
-      book = find_by(title)
-      item = Item.new(cart.item_count, book)
-      cart.items << item
-    else
-      check_item.qty += 1
+    book = find_by(title)
+    if book.stock.nil? || book.stock > 0
```
Comment (asarbu) [inline ruby_proj/app/bookstore.rb]: I think this logic should go in the Book class

File: ruby_proj/app/bookstore.rb [hunk 3]
```diff
@@ -17,12 +17,9 @@ def sort!
 
   def add_to_cart(title)
     check_item = cart.find_item(title)
```
Comment (asarbu) [inline ruby_proj/app/bookstore.rb]: I think we should rename this 🤔 `cart_item` would be better as `cart.find_item(title)` will return either a Book or nil

File: ruby_proj/app/bookstore.rb (hunk 2 above)
Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: like this? 
```
  def in_stock?
    stock.nil? || stock > 0
  end

    if book.in_stock?
      add_item_to_cart(cart_item, book)
    end
```

Comment (asarbu) [reply ruby_proj/app/bookstore.rb]: yes

Comment (iuliagatea) [reply ruby_proj/app/bookstore.rb]: done

**Pull request merged**